assets/javascripts/discourse/api-initializers/page-visits.js

```diff
@@ -0,0 +1,108 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import { ajax } from "discourse/lib/ajax";
+import { TrackedObject } from "@ember-compat/tracked-built-ins";
+import { tracked } from "@glimmer/tracking";
+import { run } from "@ember/runloop";
+import { schedule } from "@ember/runloop";
+import discourseDebounce from "discourse-common/lib/debounce";
+
+let pageVisitData = {};
+let timer = 0;
+let viewedPostIds = [];
+let screenTrack;
+let postStream;
+
+export default {
+  name: "page-visits",
+  initialize() {
+    withPluginApi("1.24.0", (api) => {
+      screenTrack = api.container.lookup("service:screen-track");
+      const currentUser = api.getCurrentUser();
+      const topicController = api.container.lookup("controller:topic");
+
+      api.onPageChange((route) => {
+        if (Object.keys(pageVisitData).length > 0) {
+          createPageVisitRecord(pageVisitData, viewedPostIds, timer);
+        }
+
+        reset();
+        startTimer();
+
+        const topicId = topicController.model?.id || null;
+        postStream = topicController.model?.postStream;
+        console.log(topicController.model);
+        if (topicId && postStream) {
+          window.addEventListener("scroll", scroll, { passive: true });
+        }
+
+        captureVisitData(currentUser?.id, topicId);
+      });
+    });
+  },
+};
+
+function scroll() {
+  discourseDebounce(this, captureOnScreenPosts, 100);
+}
+
+function captureOnScreenPosts() {
+  screenTrack._readPosts.forEach((index) => {
+    // screen track index is 1-based
+    const postId = postStream.stream[index - 1];
+    if (postId && !viewedPostIds.includes(postId)) {
+      viewedPostIds.push(postId);
+    }
+  });
+}
+
+function reset() {
+  window.removeEventListener("scroll", scroll);
+  viewedPostIds = [];
+  clearInterval(timer);
+  timer = 0;
+}
+
+function startTimer() {
+  setInterval(() => {
+    run(() => tickTimer());
+  }, 1000);
+}
+
+function tickTimer() {
+  timer += 1000;
+}
+
+function captureVisitData(userId, topicId) {
+  const data = {
+    userId: userId || null,
+    fullUrl: window.location.href,
+    userAgent: navigator.userAgent,
+    topicId: topicId,
+  };
+
+  pageVisitData = data;
+}
+
+function resetPageVisitData() {
+  pageVisitData = {};
+}
+
+async function createPageVisitRecord(data, postIds, timer) {
+  try {
+    await ajax("/page_visits", {
+      type: "POST",
```

Comment:
I am getting the error 

```
Routing Error

No route matches [POST] "/page_visits"
```

Looking at rails info routes, it exists as I would expect: 

<img width="1524" alt="Screenshot 2024-06-26 at 12 58 27 PM" src="https://github.com/discourse/discourse-page-visits/assets/50783505/740e9948-ad94-4dc6-aab9-566f3a4956cb">

- I tried including the controller file within the `plugin.rb`, eg `require_relative "app/controllers/discourse_page_visits/page_visits_controller"`

Not sure what the issue is here.

Reply:
`.json` perhaps?

Reply:
@CvX sadly no, tried that as well.

Reply:
Ah, I see now. You're mounting the engine like this:

```rb
mount ::DiscoursePageVisits::Engine, at: "discourse-page-visits"
```

so the path is `/discourse-page-visits/page_visits`, right?